test/space/ntuples_test.py

```diff
@@ -80,35 +80,26 @@ def _pos_array(fn):
     return np.abs(_array(fn)) + 0.1
 
 
-def _sparse_matrix(fn):
-    """Create a sparse positive definite Hermitian matrix for `fn`."""
-    nnz = np.random.randint(0, int(ceil(fn.size / 2)))
-    coo_r = np.random.randint(0, fn.size, size=nnz)
-    coo_c = np.random.randint(0, fn.size, size=nnz)
+def _dense_matrix(fn):
+    """Create a dense positive definite Hermitian matrix for `fn`."""
+
     if np.issubdtype(fn.dtype, np.floating):
-        values = np.random.rand(nnz).astype(fn.dtype)
+        mat = np.random.rand(fn.size, fn.size).astype(fn.dtype)
     elif np.issubdtype(fn.dtype, np.integer):
-        values = np.random.randint(0, 10, nnz).astype(fn.dtype)
+        mat = np.random.randint(0, 10, (fn.size, fn.size)).astype(fn.dtype)
```

Comment:
`astype` can take a `copy=False` option which optimizes the case of the data type returned by the `np.random` functions.